freqtrade/constants.py

```diff
@@ -199,7 +199,7 @@
                 'ccxt_config': {'type': 'object'},
                 'ccxt_async_config': {'type': 'object'}
             },
-            'required': ['name', 'key', 'secret', 'pair_whitelist']
+            'required': ['pair_whitelist']
```

Comment:
i think "name" should be required and not have a default - a user should make a concious decision on what exchange he wan'ts to use. 
I agree on the rest though.

Reply:
Agree, Removed the default value and made it required back. That was just an example of usage the new feature. It was quite painful to select first options for illustration here since changes to most of them are not logged properly in the logs.

Docs were also reviewed and slightly adjusted (removed 'bittrex' as default value, etc.),

Maybe an additional test for config without this 'name' option (the bot fails to load config) should be added later.